### PR TITLE
feat(design/autoware-architecture/sensing): add radar sensing document

### PR DESCRIPTION
**Pull request by scepter914**
## Description

This PR add radar sensing document.

The contents of the documents are as below.

- sensing/data-types/radar-data/radar-objects-data
- sensing/data-types/radar-data/radar-pointcloud-data

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by YoshiRi — APPROVED** [review]
LGTM

File: docs/design/autoware-architecture/sensing/data-types/radar-data/radar-objects-data.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+- The radar pre-process package filter noise through the `ros-perception/radar_msgs/msg/RadarTrack.msg` message type with sensor coordinate.
+- It is recommended to change the coordinate system from each sensor to base_link with message converter.
+- If there are multiple radar objects, the object merger package concatenate these objects.
+
+## Input
+
+Autoware uses radar objects data type as [radar_msgs/msg/RadarTracks.msg](https://github.com/ros-perception/radar_msgs/blob/ros2/msg/RadarTracks.msg).
+In detail, please see [Data message for radars](supported-functions/data-message.md).
+
+## Supported functions
```
Comment (miursh) [inline docs/design/autoware-architecture/sensing/data-types/radar-data/radar-objects-data.md:30]: [IMO]
"Reference implementation" might be better?

Comment (scepter914) [reply docs/design/autoware-architecture/sensing/data-types/radar-data/radar-objects-data.md]: Fixed at
https://github.com/autowarefoundation/autoware-documentation/pull/518/commits/03ba9be04565ebc102a0619a59c265c89e5b4b38
https://github.com/autowarefoundation/autoware-documentation/pull/518/commits/fab964c17a773289122fb08ae59fbc5a87d5d0b5

File: docs/design/autoware-architecture/sensing/data-types/radar-data/reference-implementations/device-driver.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Device driver for radars
+
+## Interface for radar devices
+
+The radar driver converts the communication data from radars to ROS2 topics.
```
Comment (Shin-kyoto) [inline docs/design/autoware-architecture/sensing/data-types/radar-data/reference-implementations/device-driver.md:5]:
```suggestion
The radar driver converts the communication data from radars to ROS 2 topics.
```

**Review by Shin-kyoto — CHANGES_REQUESTED** [review]

File: docs/design/autoware-architecture/sensing/data-types/radar-data/radar-objects-data.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Autoware uses radar objects data type as [radar_msgs/msg/RadarTracks.msg](https://github.com/ros-perception/radar_msgs/blob/ros2/msg/RadarTracks.msg).
+In detail, please see [Data message for radars](supported-functions/data-message.md).
+
+## Reference implementations
+
+### Device driver for radars
+
+Autoware support `ros-perception/radar_msgs/msg/RadarScan.msg` and `autoware_auto_perception_msgs/msg/TrackedObjects.msg` for Radar drivers.
+
+In detail, please see [Device driver for radars](supported-functions/device-driver.md).
```
Comment (Shin-kyoto) [inline docs/design/autoware-architecture/sensing/data-types/radar-data/radar-objects-data.md:36]:
```suggestion
In detail, please see [Device driver for radars](reference-implementations/device-driver.md).
```

File: docs/design/autoware-architecture/sensing/data-types/radar-data/reference-implementations/data-message.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+float32[6] acceleration_covariance          # Upper-triangle covariance about the x, y, z axes
+float32[6] size_covariance                  # Upper-triangle covariance about the x, y, z axes
+```
+
+### Message usage for RadarTracks
+
+- Object classifications
+
+For object classifications [radar_msgs/msg/RadarTrack.msg](https://github.com/ros-perception/radar_msgs/blob/ros2/msg/RadarTrack.msg), additional vendor-specific classifications are permitted starting from 32000.
+In addition to this, considering [existing label definition](https://gitlab.com/autowarefoundation/autoware.auto/autoware_auto_msgs/-/blob/master/autoware_auto_perception_msgs/msg/ObjectClassification.idl), Autoware define object classifications as below.
```
Comment (Shin-kyoto) [inline docs/design/autoware-architecture/sensing/data-types/radar-data/reference-implementations/data-message.md:67]:
```suggestion
Classes not defined in [radar_msgs/msg/RadarTrack.msg](https://github.com/ros-perception/radar_msgs/blob/ros2/msg/RadarTrack.msg) like vendor-specific classifications can be added by using uint64 starting from 32000.
Considering [existing label definition](https://gitlab.com/autowarefoundation/autoware.auto/autoware_auto_msgs/-/blob/master/autoware_auto_perception_msgs/msg/ObjectClassification.idl), Autoware defines object classifications as below.
```

File: docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+![radar-pointcloud-sensing](image/radar-pointcloud-sensing.drawio.svg)
+
+### Interface
+
+- Input
+  - Radar data from device
+  - Twist information of ego vehicle motion
+- Output
+  - Dynamic radar pointcloud (`ros-perception/radar_msgs/msg/RadarScan.msg`)
```
Comment (Shin-kyoto) [inline docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md:17]: What is Dynamic radar pointcloud?

File: docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+Autoware support `ros-perception/radar_msgs/msg/RadarScan.msg` and `autoware_auto_perception_msgs/msg/TrackedObjects.msg` for Radar drivers.
+
+In detail, please see [Device driver for radars](reference-implementations/device-driver.md).
+
+### Basic noise filter
+
+- [radar_threshold_filter](https://github.com/autowarefoundation/autoware.universe/tree/main/sensing/radar_threshold_filter)
+
+This package removes pointcloud noise with low amplitude, edge angle, too near pointcloud by threshold.
```
Comment (Shin-kyoto) [inline docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md:42]: [NITS]

Please add the reason for using this filter(e.g. the reason why too near pointcloud should be removed.)

File: docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+### Filter to static/dynamic pointcloud
+
+- [radar_static_pointcloud_filter](https://github.com/autowarefoundation/autoware.universe/tree/main/sensing/radar_static_pointcloud_filter)
+
+This package extract static/dynamic radar pointcloud by using doppler velocity and ego motion.
```
Comment (Shin-kyoto) [inline docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md:48]:
```suggestion
This package extracts static/dynamic radar pointcloud by using doppler velocity and ego motion.
```

File: docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+The static radar pointcloud use for localization like NDT scan matching, and the dynamic radar pointcloud use for dynamic object detection.
```
Comment (Shin-kyoto) [inline docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md:49]:
```suggestion
The static radar pointcloud can be used for localization like NDT scan matching, and the dynamic radar pointcloud can be used for dynamic object detection.
```

File: docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md (hunk 5 above)
Comment (scepter914) [reply docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md:17]: It means that pointcloud which is not static.
To express brevity for interface documentation and I think who read this document for the end can understand this word, so I would not like to change it.

File: docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md (hunk 6 above)
Comment (scepter914) [reply docs/design/autoware-architecture/sensing/data-types/radar-data/radar-pointcloud-data.md:42]: Fixed df7efc6

File: docs/design/autoware-architecture/sensing/data-types/radar-data/reference-implementations/data-message.md (hunk 4 above)
Comment (scepter914) [reply docs/design/autoware-architecture/sensing/data-types/radar-data/reference-implementations/data-message.md]: Fixed df7efc6

**Review by miursh — APPROVED** [review]
LGTM

**Review by Shin-kyoto — APPROVED** [review]

**Pull request merged**